src/web/app/components/response-time-series-chart/response-time-series-chart.component.ts

```diff
@@ -0,0 +1,129 @@
+    durationMinutes: 60,
+    intervalMinutes: 1
+  }
+
+  interval: BehaviorSubject<number> = new BehaviorSubject<number>(1);
+
+  constructor(private feedbackResponseStatsService: FeedbackResponseStatsService) { }
+
+  ngOnInit(): void {
+    this.interval.pipe(
+        switchMap(value => interval(value * 1000)), // in seconds (for easier testing), actual one will be in minutes
+        tap(() => this.refresh())
+    ).subscribe();
+  }
+
+  refresh() {
+    this.feedbackResponseStatsService.loadResponseStats(this.model.durationMinutes.toString(), this.model.intervalMinutes.toString())
+        .subscribe((data: FeedbackResponseStats) => {
+          console.log(data);
+          // parse data
+          const parsedData = this.parseData(data);
+          // draw chart
+          this.drawChart(parsedData, this.model.durationMinutes * 60 * 1000);
+        }, (resp: ErrorMessageOutput) => {
+          console.log((resp.error));
+        });
+  }
+
+  /**
+   * Handles a change in duration
+   */
+  setDurationHandler(duration: number): void {
+    this.model.durationMinutes = duration;
+  }
+
+  /**
+   * Handles a change in interval
+   */
+  setIntervalHandler(interval: number): void {
+    this.model.durationMinutes = interval;
+    this.interval.next(interval);
+  }
+
+  drawChart(data: object[], duration: number): void {
+
+    const svgWidth: number = 800;
+    const svgHeight: number = 400;
+    const margin = { top: 20, right: 20, bottom: 30, left: 50 };
+    const width: number = svgWidth - margin.left - margin.right;
+    const height: number = svgHeight - margin.top - margin.bottom;
+
+    var svg = d3.select('svg')
+        .attr("width", svgWidth)
+        .attr("height", svgHeight);
+
+    var g = svg.append("g")
+        .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+
+    var x = d3.scaleTime()
+        .rangeRound([0, width])
+        .nice();
+
+    var y = d3.scaleLinear()
+        .rangeRound([height, 0]);
+
+    x.domain([Date.now() - duration, Date.now()]);
+    y.domain(d3.extent(data, (d: any) => d.resCount));
+
+    var line = d3.line()
+        .defined((d: any) => d.timestamp >= Date.now() - duration && d.timestamp <= Date.now())
+        .x((d: any) => x(d.timestamp))
+        .y((d: any) => y(d.resCount));
+
+    g.append("g")
+        .attr("transform", "translate(0," + height + ")")
+        .call(d3.axisBottom(x));
+
+    g.append("g")
+        .call(d3.axisLeft(y))
+        .append("text")
+        .attr("fill", "#000")
+        .attr("transform", "rotate(-90)")
+        .attr("y", 6)
+        .attr("dy", "0.71em")
+        .attr("text-anchor", "end")
+        .text("No. of responses");
+
+    g.append("path")
+        .datum(data)
+        .attr("fill", "none")
+        .attr("stroke", "steelblue")
+        .attr("stroke-linejoin", "round")
+        .attr("stroke-linecap", "round")
+        .attr("stroke-width", 1.5)
+        .attr("d", line);
+  }
+
+  parseData(data: any): any[] {
```

Comment:
Try not to use any type. Although now we are in development phase, it might be acceptable.

Reply:
actually im not sure what should be the type, the data is a json object

Reply:
you can create an interface for it right?

Reply:
^ same as above, though it depends on the generated API file. If we do not have breaking change, you can first make your own interface and tryna integrate later. IDE provides powerful refactoring feature to help you with that. Defining interfaces is idiomatic in TEAMMATES, and leave any is kinda bad (defeating the point of typescript although it's a default in Angular).

Reply:
I think i can use the existing `FeedbackResponseRecord` and `FeedbackResponseRecords` interface, but I need some changes in the backend to return the json object in the form so that I can directly feed it to d3:

<img width="320" alt="Screenshot 2021-01-25 at 12 03 04 AM" src="https://user-images.githubusercontent.com/59337195/105635938-b3d82f80-5ea0-11eb-8589-6ab407a8009b.png">

Reply:
@daongochieu2810 where would you declare the interface though?